src/pudl/analysis/allocate_net_gen.py

```diff
@@ -1598,6 +1605,85 @@ def adjust_energy_source_codes(
     return gens
 
 
+def add_missing_energy_source_codes_to_gens(gens_at_freq, gf):
+    """Adds energy_source_codes that appear in the `gf` table but not `gens` to `gens`.
+
+    In some cases, non-zero fuel consumption and net generation is reported in the
+    EIA-923 generation and fuel table that is associated with an energy_source_code that
+    is not associated with that plant-prime mover in the gens table, which would cause
+    these data to get dropped when these two tables are merged. To fix this, for each
+    plant-pm, this function identifies such esc, and adds them to the `gens_at_freq`
+    table as new energy_source_code columns.
+    """
+
+    missing_gf_escs_from_gens = identify_missing_gf_escs_in_gens(gens_at_freq, gf)
+
+    # pivot these data to become new numbered energy_source_code_n columns starting at 7
```

Comment:
It starts at 7 because there are 6 defined in the gens table, right? If so, it could be nice to explain that here.

Reply:
That's correct. The original energy_source_code are numbered 1-6